manager/src/main/java/com/redhat/service/bridge/manager/connectors/ConnectorsOrchestrateImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.redhat.service.bridge.manager.connectors;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+import com.redhat.service.bridge.infra.models.dto.ConnectorStatus;
+import com.redhat.service.bridge.manager.dao.ConnectorsDAO;
+import com.redhat.service.bridge.manager.models.ConnectorEntity;
+
+import io.smallrye.mutiny.Uni;
+import io.vertx.mutiny.core.eventbus.EventBus;
+import io.vertx.mutiny.core.eventbus.Message;
+
+@ApplicationScoped
+public class ConnectorsOrchestrateImpl implements ConnectorsOrchestrate {
+
+    @Inject
+    ConnectorsDAO connectorsDAO;
+
+    @Inject
+    EventBus bus;
+
+    @Override
+    @Transactional(Transactional.TxType.NEVER)
+    public Uni<List<Message<ConnectorEntity>>> updatePendingConnectors() {
+        // assume this class is not a singleton as we could have multiple pods running this
```

Comment:
@lucamolteni I assume this is where the conversation around "locking" came in e.g we have two pods, both pick up the need to deploy the connector at the same time and therefore we need to ensure only 1 does the work.

The alternative here is that we use a Kafka Topic rather than our DB. So the need to deploy a Connector is an Event that we place onto a Kafka Topic. All our consumers of that Topic have the same consumer group id so only one of them ever consumes the Event. That would mean that we could have many pods processing work to deploy connectors, and only one of them ever doing the actual DB update when things have completed.